internal/service/enrollmentrequest.go

```diff
@@ -296,16 +304,20 @@ func (h *ServiceHandler) ApproveEnrollmentRequest(ctx context.Context, name stri
 		}
 	}
 	_, err = h.store.EnrollmentRequest().UpdateStatus(ctx, orgId, enrollmentReq)
-	return approvalStatusToReturn, StoreErrorToApiStatus(err, false, api.EnrollmentRequestKind, &name)
+	status := StoreErrorToApiStatus(err, false, api.EnrollmentRequestKind, &name)
+	h.CreateEvent(ctx, GetResourceApprovedEvent(ctx, api.EnrollmentRequestKind, name, status, h.log))
+	return approvalStatusToReturn, status
 }
 
 func (h *ServiceHandler) ReplaceEnrollmentRequestStatus(ctx context.Context, name string, er api.EnrollmentRequest) (*api.EnrollmentRequest, api.Status) {
 	orgId := store.NullOrgId
 
-	AddStatusIfNeeded(&er)
+	addStatusIfNeeded(&er)
 
 	result, err := h.store.EnrollmentRequest().UpdateStatus(ctx, orgId, &er)
-	return result, StoreErrorToApiStatus(err, false, api.EnrollmentRequestKind, &name)
+	status := StoreErrorToApiStatus(err, false, api.EnrollmentRequestKind, &name)
+	h.CreateEvent(ctx, GetResourceCreatedOrUpdatedEvent(ctx, false, api.EnrollmentRequestKind, name, status, nil, h.log))
```

Comment:
No need for events on status updates - but you should have events for ReplaceEnrollmentRequest and PatchEnrollmentRequest.

Reply:
Removed from ReplaceEnrollmentRequestStatus.
There is already events in ReplaceEnrollmentRequest and PatchEnrollmentRequest.